Modules/_asynciomodule.c

```diff
@@ -962,14 +962,18 @@ FutureObj_dealloc(PyObject *self)
 {
     FutureObj *fut = (FutureObj *)self;
 
+    PyObject_GC_UnTrack(self);
+
     if (Future_CheckExact(fut)) {
         /* When fut is subclass of Future, finalizer is called from
          * subtype_dealloc.
          */
+         PyObject_GC_Track(self);
```

Comment:
I'm just following what was done in https://bugs.python.org/issue26617. I think it's worth someone grepping through the whole codebase, I saw several other places that maybe needed this.  It seems like every dealloc should have a PyObject_GC_UnTrack at the top?